### PR TITLE
[css-overflow] property parsing and serialization

### PR DESCRIPTION
**Pull request by ewilligers**
Test that CSS Overflow properties accept the required syntax
https://drafts.csswg.org/css-overflow/#property-index

### CODE REVIEW

**Comment by ewilligers** [comment]
Blink only supports
- overflow
- overflow-x
- overflow-y
- text-overflow

It does not support overflow's `clip` keyword.

File: css/css-overflow/parsing/text-overflow-invalid.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<meta name="assert" content="text-overflow supports only the grammar 'clip | ellipsis'.">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/css/support/parsing-testcommon.js"></script>
+</head>
+<body>
+<script>
+test_invalid_value("text-overflow", 'auto');
+
+test_invalid_value("text-overflow", 'clip ellipsis');
```
Comment (frivoal) [inline css/css-overflow/parsing/text-overflow-invalid.html]: This one is valid according to level 4 syntax: https://drafts.csswg.org/css-overflow-4/#text-overflow

File: css/css-overflow/parsing/overflow-valid.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+test_valid_value("overflow", 'clip');
+test_valid_value("overflow", 'scroll');
+test_valid_value("overflow", 'auto');
+
+test_valid_value("overflow", 'visible visible', 'visible');
+test_valid_value("overflow", 'hidden visible');
+test_valid_value("overflow", 'clip clip', 'clip');
+test_valid_value("overflow", 'scroll auto');
+test_valid_value("overflow", 'auto auto', 'auto');
+
```
Comment (frivoal) [inline css/css-overflow/parsing/overflow-valid.html]: This is all correct, but you should also check this sentence:

> Computed value: as specified, except with visible/clip computing to auto/hidden (respectively) if one of overflow-x or overflow-y is neither visible nor clip


```suggestion
test_valid_value("overflow", 'auto visible', 'auto');
test_valid_value("overflow", 'visible auto', 'auto');
test_valid_value("overflow", 'hidden visible', 'hidden auto');
test_valid_value("overflow", 'visible hidden', 'auto hidden');
test_valid_value("overflow", 'scroll visible', 'scroll auto');
test_valid_value("overflow", 'visible scroll', 'auto scroll');

test_valid_value("overflow", 'auto clip', 'auto hidden');
test_valid_value("overflow", 'clip auto', 'hidden auto');
test_valid_value("overflow", 'hidden clip', 'hidden');
test_valid_value("overflow", 'clip hidden', 'hidden');
test_valid_value("overflow", 'scroll clip', 'scroll hidden');
test_valid_value("overflow", 'clip scroll', 'hidden scroll');
```

**Review by frivoal — CHANGES_REQUESTED** [review]
Mostly good, with one omission and one mistake.

File: css/css-overflow/parsing/text-overflow-invalid.html (hunk 1 above)
Comment (ewilligers) [reply css/css-overflow/parsing/text-overflow-invalid.html]: Updated the test to 'clip ellipsis clip', which remains invalid.

File: css/css-overflow/parsing/overflow-valid.html (hunk 2 above)
Comment (frivoal) [reply css/css-overflow/parsing/overflow-valid.html]: You're right, my bad. Yes, please add this in the computed value TC.

**Review by frivoal — APPROVED** [review]

File: css/css-overflow/parsing/overflow-valid.html (hunk 2 above)
Comment (ewilligers) [reply css/css-overflow/parsing/overflow-valid.html]: Computed value tests: https://github.com/web-platform-tests/wpt/pull/14923

**Pull request merged**